### DISCUSSION TITLE
pcp_combine -subtract run-total accumulation error

### DISCUSSION THREAD

**Discussion by Jeff-Duda** [dtcenter/METplus#1872] [Configuration] [answered]
I am getting a cryptic error message when trying to use `pcp_combine -subtract` to calculate 1-h precipitation from a set of GRIB2 model output files that contain run-total accumulated precip.

Command:
`pcp_combine -subtract wrftwo_conus_mem0001_03.grib2 "name='APCP'; level='A03';" wrftwo_conus_mem0001_02.grib2 "name='APCP'; level='A02';"`

A quick wgrib2 dump of the two input files reveals the precip array contents within:

> wgrib2 -v wrftwo_conus_mem0001_02.grib2 | grep "APCP"
149:142990268:d=2019060112:APCP Total Precipitation [kg/m^2]:surface:0-2 hour acc fcst:

> grib2 -v wrftwo_conus_mem0001_03.grib2 | grep "APCP"
149:142678474:d=2019060112:APCP Total Precipitation [kg/m^2]:surface:0-3 hour acc fcst:

pcp_combine fails with this output:

> DEBUG 2: Since the "-field" command line option was not used, parsing the command line arguments a list of files, each followed by a configuration string.
DEBUG 1: Reading input file: wrftwo_conus_mem0001_03.grib2
DEBUG 1: Reading data (name='APCP'; level='A03';) from input file: wrftwo_conus_mem0001_03.grib2
ERROR  : 
ERROR  : do_assign_id() -> identifier "APCP" not defined in this scope!
ERROR  : 

I do not understand a "do_assign_id" error, nor why it would be happening (higher verbosity did not output additional help). I can confirm that there is only one APCP record in both input files.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @Jeff-Duda, wanted to respond right away to this one since it should be pretty easy to fix. Unfortunately, this is just a headache caused by combining double and single quotes on the command line. Please try reversing how you're using them, as shown below.

Current command:
```
pcp_combine -subtract wrftwo_conus_mem0001_03.grib2 "name='APCP'; level='A03';" wrftwo_conus_mem0001_02.grib2 "name='APCP'; level='A02';"
```
New command:
```
pcp_combine -subtract wrftwo_conus_mem0001_03.grib2 'name="APCP"; level="A03";' wrftwo_conus_mem0001_02.grib2 'name="APCP"; level="A02";'
```
Anything inside single quotes is treated as a literal string and the shell does no additional interpretation of what's inside. The MET config file parser expects strings to be enclosed in double-quotes `"`. And these command line option strings are parsed as mini-config files. So enclosing each one in single quotes should do the trick.

If not, please let me know.

**Reply by Jeff-Duda** [reply]
Perfect. Thanks for the quick response, John!
